pkg/svc/zserver.go

```diff
@@ -2,51 +2,123 @@ package svc
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
 const (
 	// version is the current version of the service
-	version = "0.0.1"
+	version  = "0.0.1"
+	Internal = "Internal error"
 )
 
 // Default implementation of the MyResponder server interface
-type server struct{}
+type server struct {
+	pb.UnimplementedMyResponderServer
+	Description string
+	Timestamp   time.Time
+	Requests    int64
+	pubsub      *dapr.PubSub
+}
 
 // GetVersion returns the current version of the service
 func (server) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
 	return &pb.VersionResponse{Version: version}, nil
 }
 
+func (s *server) GetDescription(ctx context.Context, req *pb.GetDescriptionRequest) (*pb.GetDescriptionResponse, error) {
+	if req.GetService() != 2 {
+		data := []byte("ping")
+		s.pubsub.Publish("info", data)
+		return &pb.GetDescriptionResponse{Description: <-s.pubsub.Buffer}, nil
+	}
+	s.Requests += 1
+	return &pb.GetDescriptionResponse{Description: s.Description}, nil
+}
+
+func (s *server) UpdateDescription(ctx context.Context, req *pb.UpdateDescriptionRequest) (*pb.UpdateDescriptionResponse, error) {
+	if req.GetDescription() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Description can't be empty")
+	}
+	if req.GetService() != 2 {
+		data := []byte(req.Description)
+		s.pubsub.Publish("update-info", data)
+		return &pb.UpdateDescriptionResponse{Description: <-s.pubsub.Buffer}, nil
+	}
+	s.Requests += 1
+	s.Description = req.Description
+	return &pb.UpdateDescriptionResponse{Description: s.Description}, nil
+}
+
+func (s *server) GetUptime(ctx context.Context, req *pb.GetUptimeRequest) (*pb.GetUptimeResponse, error) {
+	if req.GetService() != 2 {
+		data := []byte("ping")
+		s.pubsub.Publish("uptime", data)
+		i, err := strconv.Atoi(<-s.pubsub.Buffer)
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetUptimeResponse{Uptime: int64(i)}, nil
+	}
+	s.Requests += 1
+	uptime := time.Now().Unix() - s.Timestamp.Unix()
+	return &pb.GetUptimeResponse{Uptime: uptime}, nil
+}
+
+func (s *server) GetRequests(ctx context.Context, req *pb.GetRequestsRequest) (*pb.GetRequestsResponse, error) {
+	if req.GetService() != 2 {
+		data := []byte("ping")
+		s.pubsub.Publish("requests", data)
+		i, err := strconv.Atoi(<-s.pubsub.Buffer)
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetRequestsResponse{Requests: int64(i)}, nil
+	}
+	s.Requests += 1
+	return &pb.GetRequestsResponse{Requests: s.Requests}, nil
+}
+func (s *server) GetMode(ctx context.Context, req *pb.GetModeRequest) (*pb.GetModeResponse, error) {
+	data := []byte("ping")
+	s.pubsub.Publish("get-mode", data)
+	i, err := strconv.Atoi(<-s.pubsub.Buffer)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.GetModeResponse{Mode: int64(i)}, nil
+}
+
+func (s *server) SetMode(ctx context.Context, req *pb.SetModeRequest) (*pb.SetModeResponse, error) {
+	data := []byte("ping")
```

Comment:
Not sure I understand how will Storage know what value of mode it needs to set...

Reply:
Fixed